src/term.rs

```diff
@@ -445,6 +445,32 @@ impl Term {
             _ => false,
         }
     }
+
+    /// Returns `true` if `self` has any free vairables.
+    ///
+    /// # Example
+    /// ```
+    /// use lambda_calculus::*;
+    ///
+    /// let with_freevar = abs(Var(2));     // λ 2
+    /// let without_freevar = abs(Var(1));  // λ 1
+    ///
+    /// assert!(with_freevar.has_free_variables());
+    /// assert!(!without_freevar.has_free_variables());
+    pub fn has_free_variables(&self) -> bool {
+        self.has_free_variables_helper(0)
+    }
+
+    fn has_free_variables_helper(&self, depth: usize) -> bool {
+        match self {
+            Var(x) => *x > depth || *x == 0,
+            Abs(p) => p.has_free_variables_helper(depth + 1),
+            App(p_boxed) => {
+                let (ref f, ref a) = **p_boxed;
+                f.has_free_variables_helper(depth) && a.has_free_variables_helper(depth)
```

Comment:
I'm a bit rusty with lambda calculus these days, but wouldn't it suffice for either of these (sub)terms to have a free variable for the entire term to be considered as having them?

Reply:
That's true - line 470 should have an `||`, d'oh! I was only using the function on expressions with long abstraction chains, so I must've missed this.